[WIP] Add rates to pricing

dlapiduz opened the pull request:
Lets add a rates page to pricing. This is a first stab lets chat about it.

/cc @mogul @NoahKunin @afeld @wslack

dlapiduz committed:
Add rates to pricing

afeld commented on content/intro/pricing/rates.md:
```diff
@@ -0,0 +1,89 @@
+weight: 10
+---
+
+## cloud.gov is charged with 3 different components:
```
Minor: I don't think this needs to be a heading.

afeld commented on content/intro/pricing/rates.md:
```diff
@@ -0,0 +1,89 @@
+    <th>$60K / year</th>
+    <th>$90K / year</th>
+  </tr>
+</table>
```
I'd move the table above the list of what's included, since this is the top-line thing that visitors will care about.

afeld replied:
Maybe there should be a separate "What's included" section?

wslack replied:
Could go either way, but I think having a single page for "what do I get for my money?" is helpful.

dlapiduz replied:
One of our biggest challenges is making sure people don't compare this to other public PaaSes as apples to apples since it isn't. The PaaS is just a hack to get us through the compliance part.

So I think keeping at least some of this is critical to the purpose of this page...

afeld commented on content/intro/pricing/rates.md:
```diff
@@ -0,0 +1,89 @@
+
+The price of each service is different and they are billed by the service instance. You can see the price of each service in the [marketplace](#) (coming soon).
+
+Services instances are billed per quarter, tallied monthly and serverable.
```
If we're not currently charging for services, can we make it clear that this will be a change?

afeld commented on content/intro/pricing/rates.md:
```diff
@@ -0,0 +1,89 @@
+
+Managed services are those which cloud.gov will spin up instantly and run on your behalf. You pay for the resources they consume and our efforts to provide them as a service.
+
+The price of each service is different and they are billed by the service instance. You can see the price of each service in the [marketplace](#) (coming soon).
```
_How_ soon? I would skip mentioning the Marketplace (except maybe in a Roadmap) until it's ready.

afeld commented on content/intro/pricing/rates.md:
```diff
@@ -0,0 +1,89 @@
+
+### 2. cloud.gov Usage Rates
+
+Resources such as memory, storage and CPU are allocated as needed to run your applications. You will be charged for the resources your applications consume, as a factor of the memory used.
```
I don't think these sentences are necessary, or at most they could be moved to a separate page. ~~I'd also link to/from the [quotas](https://docs.cloud.gov/apps/quotas/) page.~~

afeld commented on content/intro/pricing/rates.md:
```diff
@@ -0,0 +1,89 @@
+
+Resources such as memory, storage and CPU are allocated as needed to run your applications. You will be charged for the resources your applications consume, as a factor of the memory used.
+
+<em>Charged at $0.0033 per MB reserved per hour (<b>~$99 / GB / month</b>), billed per quarter, tallied monthly, severable.</em>
```
I'd add "[Learn more about quotas](https://docs.cloud.gov/apps/quotas/)."

wslack replied:
The Quotas page will need an update (or a section for internal vs external users); right now it lacks instructions for a quota change.

afeld commented on content/intro/pricing/rates.md:
```diff
@@ -0,0 +1,89 @@
+
+---
+
+Note: This pricing includes any services created through cloud.gov. 18F Infrastructure might offer/bill additional IaaS services separately depending on your project needs.
```
It might be overly cautious, but I would be more explicit here that we would only do that if they requested/agreed to it...we wouldn't just bill them for extra services on a whim.

wslack replied:
I think this can be a link to a general IaaS page, with text like "You may also set up additional IaaS support with 18F with separate pricing."

mogul replied:
We specifically opted to steer them away from just skipping cloud.gov and asking for AWS services.

afeld commented on content/intro/pricing/rates.md:
```diff
@@ -0,0 +1,89 @@
+- TCO statement (red/not-included stuff)
+- Link to terminology page
+- Add examples
+ -->
```
Unless you plan to do these within the same pull request, mind moving them to an issue/card?

wslack commented on content/intro/pricing/rates.md:
```diff
@@ -0,0 +1,89 @@
+      For .gov and .mil accounts to kick the tires. Capped usage, all managed services available, no paperwork required.
+    </td>
+    <td>
+      All the gov-facing services you want on apps.cloud.gov, low separation between teams. Usually one per agency/department. Sandbox caps are removed.
```
"Low separation between teams" is an example of the sort of language unclear to non-tech visitors.

mogul committed:
Consistent terminology

mogul committed:
Assorted clarifications

mogul committed:
Split org/app quota info

The org quota info is much more germane to pricing/billing, whereas the
application quota information is specifically about deploying apps, and
should remain in that section.

mogul committed:
Consistent terminology

mogul committed:
Assorted fixes

- Break out what is included to a new page.
- Move sandbox usage cap info into the usage section
- Better info about how to find service pricing info for now

mogul committed:
Better managed service info

mogul committed:
Be more direct about non-cloud.gov services

mogul committed:
Minor punctuation clean-up

mogul committed:
Move provisional status to the bottom, clean up wording

mogul committed:
Make the sections parallel

mogul committed:
merge from master

dlapiduz commented on content/apps/quotas.md:
```diff
@@ -2,64 +2,14 @@
   main:
     parent: apps
-title: Quotas
+title: Application Quotas
```
What are application quotas?

mogul replied:
Just what it says in the page... These are the specific settings for `memory`, `disk_quota`, and `instances` that you'd typically put in your manifest, or specify via `cf` with flags `-m`, `-k`, or `-i`. I suppose we need yet another page for `space-quota`.

dlapiduz replied:
Do you mind if we change it to `Application Limits`? `Application Quotas` is a non standard term.

dlapiduz commented on content/intro/pricing/rates.md:
```diff
@@ -0,0 +1,94 @@
+- Usage
+- Managed services
+
+[Find out more about this pricing model.]({{< relref "intro/pricing/pricing-model.md" >}})
```
So we are basically moving "What's included" 2 levels out? I think we should have something here.

Also, if it takes 3 pages to explain how we charge I think there is something broken.

dlapiduz replied:
Oh, just saw the "whats included" link below. I would remove this link then.

mogul replied:
The pricing model page still has a lot of related info on it that doesn't have a better home, eg that it's a cost-recovery model, that it's still in flux and needs feedback, that we want people to think about the money it's saving them in personnel and compliance delays, etc. I think I want to slim down the pricing model page to _just_ focus on those points, and leave the rest of the model definition to the terminology page.

dlapiduz replied:
That makes sense... lets do it

mogul committed:
Reserve "quota" terminology for orgs, apps have "limits"

mogul committed:
Fix broken links

mogul committed:
Distill definitions into one page and reference it

mogul committed:
Identify the value proposition, discourage apple/oranges comparison

mogul committed:
Use Hugo's anchor-generation facility

dlapiduz committed:
Remove the concept of "Quota Unit"

dlapiduz committed:
Fix typo

NoahKunin committed:
A la carte doesn't exist yet

Therefore it shouldn't be listed

NoahKunin committed:
Style edits, pricing stability clarification, and extraneous info clean up

Ellipses in a paragraph is not a style we should adopt in official docs. 

Also, we make use of power provided by people other than "men". Remember to always use the full name of a thing before using the acronym on a single page. ASF info is extraneous and does not provide value (also technically incorrect to refer it to a grant, which has a technical definition, even if it semi-functions as a grant).

"A la carte" items are not within cloud.gov pricing at the moment, but the broader Infrastructure charge. Once cloud.gov's tooling actually contain these costs, we can list them here, but we also aren't explaining them yet, so listing them is confusing. Also, when we do, we should not use French to describe them. #politicalreality

Other important stylistic edits - the audience for this doc is already ingesting a tremendous amount of new information. Let's keep it to what they need to know, not our internal baseball.

NoahKunin committed:
important point about wcfs

NoahKunin committed:
Clarity on sandbox and cleanup edit

There was a question in doc about what the Sandbox is capped at. Calling it 1GB for now until someone tells me different. Also, it is *very* confusing  to ever say "unrecovered costs*. All costs are recovered. The Sandbox is recovered through upstream funding and sales.

Consistent term usage from the other pages, consistent capitalization, unnecessary formatting.

Skunkworks is never defined, and while a highly relevant concept to us, not a term for a general audience without clarification. "Prototyping" aligns with how that plan is usually requested, and how we talk about our work in that phase in other contexts.

NoahKunin committed:
important clarification about funding

Funds themselves are not severable or non-severable. Services are severable or non-severable, inherent to themselves. The important variable  to look at with federal funding is its "color", otherwise understood to be it's appropriation length (i.e. 1 year, 2 year, or no-year / indefinite). The color limits the time frame in which the money can be spent, regardless of the nature of the service or commodity. 

Since a non-severable and non-refundable services is actualized the moment a bona fide need is identified and the service is paid for, it does not expire at the end of the time frame of the underlying appropriation, because it was already _spent_ correctly according to its color.

wslack commented on content/intro/pricing/pricing-model.md:
```diff
@@ -8,45 +8,25 @@ weight: 10
+Comparing cloud.gov's pricing model to those private Platform as a Service (PaaS) providers is extremely difficult, as it is not an apples to apples comparison. If you attempt to do a price comparison, be sure to include the up-front and opportunity costs of your procurement, operations, security, and compliance teams in obtaining a solution from scratch. It may be difficult to obtain these costs ...
 
-## À la carte items
+It is critical to obtain all this data in order to perform a true cost of ownership comparison. 18F stands ready to help you build an accurate cost model.
```
We should elaborate on this, and how much time we would expect for this work. Otherwise, I worry about losing a lot of time to this.

afeld replied:
:+1: 

Maybe sufficient to say "via [18F Consulting](https://18f.gsa.gov/consulting/)"?

wslack commented on content/intro/pricing/whats-included.md:
```diff
@@ -0,0 +1,35 @@
+weight: 10
+---
+
+All cloud.gov access packages include the following: 
```
It might be good to get some plain language here. I'm thinking about someone on the $$$ side looking to understand the below.

afeld replied:
This would be a good thing to revisit with the Writing Lab.

afeld commented on content/apps/limits.md:
```diff
@@ -0,0 +1,71 @@
+---
+
+
+As a Cloud Foundry user you’re free to divide your [organization's quota]({{< relref "intro/pricing/quotas.md" >}}) capacity amongst your applications as you see fit by way of application limits. Application limits are specified in your application manifest or as `cf push` command line options.
```
Mind linking "application limits" to `{{< relref "quotas.md#application-quotas" >}}`?

afeld replied:
Ha, sorry, didn't see that this was moved below. Don't mind me!

afeld commented on content/apps/limits.md:
```diff
@@ -0,0 +1,71 @@
+
+As a Cloud Foundry user you’re free to divide your [organization's quota]({{< relref "intro/pricing/quotas.md" >}}) capacity amongst your applications as you see fit by way of application limits. Application limits are specified in your application manifest or as `cf push` command line options.
+
+## Application Limit Options:
```
Minor: mind using sentence case for the headings and removing the trailing colons?

https://pages.18f.gov/content-guide/capitalization/#headings

afeld commented on content/apps/limits.md:
```diff
@@ -0,0 +1,71 @@
+		#0   running   2015-07-30 05:58:11 PM   0.0%   94.6M of 128M   80.4M of 128M      
+
+
+- Any application which exceeds its memory limit will be automatically restarted. Use `cf events APPNAME` to look for 'out of memory' crashes.
```
Not a blocker, but we should see if we can this page merged upstream to the canonical Cloud Foundry docs.

afeld commented on content/apps/limits.md:
```diff
@@ -0,0 +1,71 @@
+
+Sets the number of application instances to launch. Each additional instance receives the same memory and disk reservation. An application with a manifest specifying `memory: 256M` and `instances: 4` would reserve 1GB (256M x 4) total.
+
+## Application Limit Options : Memory Share equals Compute Share
```
Minor: can we change this to `### Memory share`?

afeld commented on content/apps/limits.md:
```diff
@@ -0,0 +1,71 @@
+
+As noted above, your application's compute limit is derived from a its memory limit. Each application receives a compute share equal to its relative share of memory.
+
+For example, with a 1 unit (3.75GB) quota:
```
Where does the term "unit" come from? Can we link somewhere?

afeld commented on content/intro/pricing/quotas.md:
```diff
@@ -0,0 +1,35 @@
+- Application routes.
+- Application memory.
+- Service instances.
+- Access to paid plans.
```
Paid plans, or services?

afeld commented on content/intro/pricing/quotas.md:
```diff
@@ -0,0 +1,35 @@
+
+In this situation you have three options:
+
+1. Delete existing resources with `cf delete`, `delete-service`, `delete-route` or similar.
```
Not a blocker, but would be great to link to the docs for each of these.

afeld commented on content/intro/pricing/quotas.md:
```diff
@@ -0,0 +1,35 @@
+
+1. Delete existing resources with `cf delete`, `delete-service`, `delete-route` or similar.
+2. Reconfigure individual existing [Application Quotas]({{< relref "apps/limits.md" >}}) and redeploy.
+3. Request a quota change by an administrator via a [DevOps issue](https://github.com/18F/DevOps/issues).
```
I'd just link to the [Help](https://docs.cloud.gov/help/) page, since external folks won't have access to the DevOps repo.

afeld commented on content/intro/pricing/rates.md:
```diff
@@ -0,0 +1,92 @@
+- Resource usage
+- Managed services
+
+Find out [more about this pricing model]({{< relref "intro/pricing/pricing-model.md" >}}) and [how we define these terms]({{< relref "intro/terminology/pricing-terminology.md" >}})
```
Minor: missing period at the end of the sentence.

afeld commented on content/intro/pricing/rates.md:
```diff
@@ -0,0 +1,92 @@
+
+Access package fees are charged yearly and are non-severable. 
+
+[Find more about what's included in the access package.]({{< relref "intro/pricing/whats-included.md" >}})
```
Minor: no need for the line break above.

afeld commented on content/intro/pricing/rates.md:
```diff
@@ -0,0 +1,92 @@
+- Create buildpack page with list and compliance trade-offs
+- Link to terminology page
+- Add examples
+-->
```
Mind moving these to issues/cards?

afeld commented on content/intro/pricing/whats-included.md:
```diff
@@ -0,0 +1,35 @@
+
+- API access, CLI access, Web UI access
+- Documentation/tutorial material (including continuing development)
+- Web-based support during business hours (platform available 24/7, problems addressed as-possible).
```
Minor: link to the [Help](http://docs.cloud.gov/help/) page.

afeld commented on content/intro/pricing/whats-included.md:
```diff
@@ -0,0 +1,35 @@
+- API access, CLI access, Web UI access
+- Documentation/tutorial material (including continuing development)
+- Web-based support during business hours (platform available 24/7, problems addressed as-possible).
+- Allocation of underlying IaaS (AWS) instances 
```
Maybe simplify to "underlying server resources"?

afeld commented on content/intro/pricing/whats-included.md:
```diff
@@ -0,0 +1,35 @@
+- Documentation/tutorial material (including continuing development)
+- Web-based support during business hours (platform available 24/7, problems addressed as-possible).
+- Allocation of underlying IaaS (AWS) instances 
+- OS security updates and regular hardening of OS image
```
I think this can be simplified to "Regular security updates to underlying infrastructure".

afeld replied:
...and combined with the following bullet.

afeld commented on content/intro/pricing/whats-included.md:
```diff
@@ -0,0 +1,35 @@
+- OS security updates and regular hardening of OS image
+- Regular updates to platform security
+- Network security
+- ATO-ready documentation of system components (compliance-masonry form)
```
Maybe "via [Compliance Masonry](https://github.com/opencontrol/compliance-masonry)" (with link). Also, would "ATO-ready" be overselling it a bit? Should it say something to the effect of "Documentation to ease your agency's Authority to Operate process"?

/cc @geramirez

afeld commented on content/intro/pricing/whats-included.md:
```diff
@@ -0,0 +1,35 @@
+  + go_buildpack
+  + python_buildpack
+  + php_buildpack
+  + binary_buildpack
```
Not sure it's necessary to list the buildpacks here.

afeld commented on content/intro/pricing/whats-included.md:
```diff
@@ -0,0 +1,35 @@
+  + python_buildpack
+  + php_buildpack
+  + binary_buildpack
+- Self-service management of spaces and users within an organization
```
/cc https://github.com/18F/cg-docs/pull/166

afeld commented on content/intro/terminology/pricing-terminology.md:
```diff
@@ -14,14 +14,33 @@ A "System" roughly corresponds to a product or project team, including all the i
+## Support {#support}
 
-“Support” is support for any inability to use the platform as intended, on a best-effort basis. We have no existing service-level agreement. We plan to always publish metrics that give agencies the ability to make an informed choice about whether to use cloud.gov or an alternative PaaS solution based on our track record.
+“Support” is support to use the platform as intended, on a best-effort basis. We have no existing service-level agreement. We plan to always publish metrics that give agencies the ability to make an informed choice about whether to use cloud.gov or an alternative PaaS solution based on our actual track record.
```
Maybe add

``` markdown
See the [Help]({{< relref "help.md" >}} page for information about how to get in touch.
```

afeld commented on content/intro/terminology/pricing-terminology.md:
```diff
@@ -14,14 +14,33 @@ A "System" roughly corresponds to a product or project team, including all the i
+
+Resource usage fees on-demand are understood to be “severable” as they can be controlled in a self-service manner by provisioning more or fewer apps, and therefore must be treated accordingly.
+
+Agencies can purchase reserved capacity via 18F as well, which is understood to be “non-severable”.
```
Is this different than adjusting the organization quota?

dlapiduz replied:
Oh, I see why this is confusing... Basically here you are buying reserved "reserved" capacity. As in we buy reserved instances in advance and you can use it with quotas which are "reserved" capacity.

dlapiduz replied:
We need to find better wording for this...
